src/cmd/services/m3coordinator/ingest/write.go

```diff
@@ -395,20 +423,28 @@ func (d *downsamplerAndWriter) writeAggregatedBatch(
 			}
 		}
 
-		samplesAppender, err := appender.SamplesAppender(opts)
+		result, err := appender.SamplesAppender(opts)
 		if err != nil {
-			return err
+			multiErr = multiErr.Add(err)
+		}
+
+		if result.IsDropPolicyApplied {
+			iter.SetCurrentMetadata(ts.Metadata{true})
 		}
 
 		for _, dp := range datapoints {
-			err := samplesAppender.AppendGaugeTimedSample(dp.Timestamp, dp.Value)
+			err := result.SamplesAppender.AppendGaugeTimedSample(dp.Timestamp, dp.Value)
 			if err != nil {
-				return err
+				// If we see an error break out so we can try processing the
+				// next datapoint.
```

Comment:
Hm do we really want to break out? Think it's worth trying all samples (so no individual error breaks other values).

Reply:
Yeah, makes sense.